controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -174,6 +183,71 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageConsumerReconciler) reconcileOnboardingSecret() error {
+	secret := &corev1.Secret{}
+	secret.Name = fmt.Sprintf("onboarding-token-%d", util.FnvHash(string(r.storageConsumer.UID)))
+	secret.Namespace = r.storageConsumer.Namespace
+	if err := r.get(secret); client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to get secret %s: %v", secret.Name, err)
+	}
+
+	if !r.storageConsumer.Spec.Enable {
+		// if not enabled, ensure secret presence
+		if secret.UID == "" {
+			if err := r.own(secret); err != nil {
+				return err
+			}
+			storageCluster, err := util.GetStorageClusterInNamespace(r.ctx, r.Client, r.namespace)
+			if err != nil {
+				return err
+			}
+			salt := uuid.New().String()
+			token, err := util.GenerateClientOnboardingToken(
+				tokenLifetimeInHours,
+				onboardingPrivateKeyFilePath,
+				nil,
+				storageCluster.UID,
+				salt,
+			)
+			if err != nil {
+				return err
+			}
+			secret.Data = map[string][]byte{
+				"ticket": []byte(token),
+			}
+			util.AddAnnotation(secret, "token-expiry-utc", time.Now().Add(tokenLifetimeInHours*time.Hour).UTC().String())
+			// as salt is random we are annotating it on secret as well to read later if we fail to annotate consumer in current reconcile
+			util.AddAnnotation(secret, TicketAnnotation, salt)
+			r.Log.Info("creating secret containing onboarding token for", "StorageConsumer", r.storageConsumer.Name, "Secret", secret.Name)
+			if err := r.Create(r.ctx, secret); err != nil {
+				return err
+			}
+			// annotating on consumer helps us validate the identity during onboard
+			util.AddAnnotation(r.storageConsumer, TicketAnnotation, salt)
+			if err := r.Update(r.ctx, r.storageConsumer); err != nil {
+				return fmt.Errorf("failed to add ticket reference to storageconsumer %s: %v", r.storageConsumer.Name, err)
+			}
```

Comment:
If you have a way to recognize either the secret or the consumer, you should have a way to get from one to another without the need for special annotation